modules/database/export.bal

```diff
@@ -0,0 +1,485 @@
+    stream<StepDataExport, sql:Error?> stepDataResult = experimentDB->query(`SELECT dataId, relationType FROM StepData WHERE stepId = ${stepId};`);
+    StepDataExport[]|error|() stepDataList = from var stepData in stepDataResult
+        select stepData;
+    if stepDataList is error {
+        return error(string `[stepId ${stepId}] Could not retrieve step data for export.`);
+    } else if stepDataList is () {
+        return [];
+    } else {
+        return stepDataList;
+    }
+}
+
+public isolated transactional function getTimelineSubstepDataList(int stepId, int substepNr) returns SubstepDataExport[]|error {
+    stream<SubstepDataExport, sql:Error?> substepDataResult = experimentDB->query(`SELECT substepNr, dataId, relationType FROM SubstepData WHERE stepId = ${stepId} AND substepNr = ${substepNr};`);
+    SubstepDataExport[]|error|() substepDataList = from var substepData in substepDataResult
+        select substepData;
+    if substepDataList is error {
+        return error(string `[stepId ${stepId}, substepNr ${substepNr}] Could not retrieve substep data for export.`);
+    } else if substepDataList is () {
+        return [];
+    } else {
+        return substepDataList;
+    }
+}
+
+public isolated transactional function getTimelineSubstepsBaseExport(int stepId) returns TimelineSubstepExportBase[]|error {
+    stream<TimelineSubstepExportBase, sql:Error?> substepsResult = experimentDB->query(
+        `SELECT substepNr, substepId, href, hrefUi, cleared, parameters, parametersContentType FROM TimelineSubstep WHERE stepId=${stepId} ORDER BY substepNr ASC;`
+    );
+    TimelineSubstepExportBase[]|error|() substepsBase = from var substep in substepsResult
+        select substep;
+    if substepsBase is error {
+        return substepsBase;
+    } else if substepsBase is () {
+        return [];
+    } else {
+        return substepsBase;
+    }
+}
+
+public isolated transactional function getTimelineSubstepsExport(int stepId) returns TimelineSubstepExport[]|error {
+    TimelineSubstepExport[] substepsExport = [];
+    TimelineSubstepExportBase[] substepsBase = check getTimelineSubstepsBaseExport(stepId);
+    foreach var substep in substepsBase {
+        SubstepDataExport[] substepDataList = check getTimelineSubstepDataList(stepId, substep.substepNr);
+        TimelineSubstepExport substepExport = {
+            substepId: substep.substepId,
+            href: substep.href,
+            hrefUi: substep.hrefUi,
+            cleared: substep.cleared,
+            parameters: substep.parameters,
+            parametersContentType: substep.parametersContentType,
+            substepDataList: substepDataList
+        };
+        substepsExport.push(substepExport);
+    }
+    return substepsExport;
+}
+
+public isolated transactional function getExperimentDataExport(int experimentId, int dataId) returns ExperimentDataExport|error {
```

Comment:
Why not export all data by default and filter in ballerina (e.g. to not export unused data)?